src/bluesearch/database/article.py

```diff
@@ -73,6 +76,127 @@ def paragraphs(self) -> Iterable[Tuple[str, str]]:
         """
 
 
+class PubmedXMLParser(ArticleParser):
+    """Parser for PubMed XML files using the JATS Journal Publishing DTD.
+
+    Parameters
+    ----------
+    path
+        The path to the XML file from PubMed.
+    """
+
+    def __init__(self, path: Union[str, Path]) -> None:
```

Comment:
One thing to consider here is that the `CORD19ArticleParser` expects a `dict` in the constractor. Which assumes that the user already loaded the `.json` file. 

Here, we are passing a path. I think we should unify this somehow.

Personally, I would vote for passing paths everywhere. The main reason for it is that we could construct the parsers programmatically.

```python
def construct_parser(parser, path, **kwargs):
      parser_cls = getattr("bluesearch.database.article", parser)
      return parser_cls(path, **kwargs)
```

Reply:
Yes we saw that. And @Stannislav mentioned the fact that testing was more complicated if we are passing a path too. But I think he had also other arguments, right @Stannislav ?

Reply:
@Stannislav hehe, check out this #412, specifically the `add` command. TLDR the goal is to write a CLI that takes
* parser class name
* path to file to be parsed
and then internally parses that file and puts it in the database. However, if the constructor of different `ArticleParser` subclasses differs, then you would need to have a special logic for each of them

Reply:
IMO parsing should take the raw files and spit out nice Python objects.

The following examples come to mind
```
pd.read_csv(path)
pd.read_pkl(path)
```

Regarding testing, we literally have a bunch of `.json` files inside of `tests/data/...` so in a way it is simpler to just provide their paths than creating a derived fixture that `json.load` them.

I guess what might be annoying is writing a Python `dict` in the source code and then having to `json.dump` it and remember the path. However, I do not think we have tests like this.

Reply:
So I guess #419 is the followup to this?

Reply:
Yes #419 is the followup.